File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    tanCount = {}
    for i in set(tangerine):
        tanCount[i] = 0
    for i in tangerine:
        tanCount[i]+=1
    tanCount = dict(sorted(tanCount.items(), key=lambda x:x[1], reverse=True))
    count = 0
    for i in tanCount.values():
        count+=i
        answer+=1
        if count >= k:
            return answer

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    const dict = {}
    let answer = 0;
    let count = 0;
    
    for(val of tangerine) {
        if(!dict[val]) dict[val] = 1
        else {
            dict[val] +=1
        }
    }
    
    const countSort = Object.entries(dict).sort((a, b) => b[1] - a[1] )
    
    for([size, cnt] of countSort) {
        answer +=1;
        count+=cnt
        if(count >= k) return answer
    }
}